packages/types/src/translators/transaction-view.ts

```diff
@@ -0,0 +1,40 @@
+import { asOpaqueMemoView, asReceiverMemoView } from './memo-view';
+import { asPublicActionView, asReceiverActionView } from './action-view';
+import { TransactionView } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1alpha1/transaction_pb';
+import { Translator } from './types';
+import { Address } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1alpha1/keys_pb';
+
+export const asPublicTransactionView: Translator<TransactionView> = transactionView => {
+  if (!transactionView?.bodyView) return new TransactionView();
+
+  return new TransactionView({
+    bodyView: {
+      memoView: asOpaqueMemoView(transactionView.bodyView.memoView),
+      actionViews: transactionView.bodyView.actionViews.map(asPublicActionView),
+
+      ...(transactionView.bodyView.transactionParameters
+        ? { transactionParameters: transactionView.bodyView.transactionParameters }
+        : {}),
+    },
+  });
+};
+
+export const asReceiverTransactionView: (
+  isControlledAddress: (address: Address) => Promise<boolean>,
+) => Translator<TransactionView, Promise<TransactionView>> =
+  isControlledAddress => async transactionView => {
+    if (!transactionView?.bodyView) return new TransactionView();
+
+    return new TransactionView({
+      bodyView: {
+        memoView: asReceiverMemoView(transactionView.bodyView.memoView),
+        actionViews: await Promise.all(
+          transactionView.bodyView.actionViews.map(asReceiverActionView(isControlledAddress)),
+        ),
```

Comment:
There are a number of places like this where I've left out some properties like `detectionData`, since they don't get rendered to the screen in the `TransactionViewComponent` anyway. 

Are we OK with that? The risk that comes to mind is that, if we start rendering stuff like `detectionData` via `TransactionViewComponent` in the future, we'll likely forget to update that here, and data might not get rendered that should be.

Reply:
yeah, i think this is fine, because the "translators" here are scoped to the confirmation dialog -- we don't have a problem of forgetting to add data in the confirmation dialog, because the confirmation dialog is the top-level part that _defines_ what data should be shown